### PR TITLE
Fallback to default favicon

### PR DESCRIPTION
**Pull request by juliusknorr**
Respond with the default Nextcloud favicon/touchicon instead of a 404 status.

Fix for https://github.com/nextcloud/server/issues/3428

Please review @nextcloud/theming

### CODE REVIEW

**Comment by juliushaertl** [comment]
cc @caco3

**Comment by MorrisJobke** [comment]
```
curl -X HEAD http://192.168.99.100/server/apps/theming/icon/core -i
HTTP/1.1 200 OK
...
Content-Disposition: inline; filename=""
Content-Length: 2918
Content-Type: text/html; charset=UTF-8
```

* [ ] the content type is wrong

**Comment by juliushaertl** [comment]
@MorrisJobke Fixed :wink:

File: apps/theming/lib/Controller/IconController.php [hunk 1]
```diff
@@ -132,16 +135,21 @@ public function getFavicon($app = "core") {
...
-				$response->addHeader('Expires', $expires->format(\DateTime::RFC2822));
-				$response->addHeader('Pragma', 'cache');
-				return $response;
 			}
 		}
-		return new NotFoundResponse();
+		if($response === null) {
+			$fallbackLogo = \OC::$SERVERROOT . '/core/img/favicon.png';
+			/** @var FileAccessHelper */
+			$fileAccessHelper = \OC::$server->query(FileAccessHelper::class);
```
Comment (rullzer) [inline apps/theming/lib/Controller/IconController.php]: Any reason not to inject this so you can do proper unit testing?

File: apps/theming/lib/Controller/IconController.php [hunk 2]
```diff
@@ -163,15 +172,20 @@ public function getTouchIcon($app = "core") {
...
-				$response->addHeader('Expires', $expires->format(\DateTime::RFC2822));
-				$response->addHeader('Pragma', 'cache');
-				return $response;
 			}
 		}
-		return new NotFoundResponse();
+		if($response === null) {
+			$fallbackLogo = \OC::$SERVERROOT . '/core/img/favicon-touch.png';
+			/** @var FileAccessHelper */
+			$fileAccessHelper = \OC::$server->query(FileAccessHelper::class);
```
Comment (rullzer) [inline apps/theming/lib/Controller/IconController.php]: Same here

**Review by rullzer — CHANGES_REQUESTED** [review]

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

File: apps/theming/lib/Controller/IconController.php (hunk 1 above)
Comment (juliushaertl) [inline apps/theming/lib/Controller/IconController.php]: No, guess that was why i started using it :see_no_evil: 

Fixed.

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**